larray/viewer.py

```diff
@@ -2295,16 +2296,34 @@ def on_item_changed(self, curr, prev):
             else:
                 self.eval_box.setText(expr)
 
-    # TODO: rename to set_current_array
-    def set_widget_array(self, array, title):
+    def update_title(self):
+        current_item = self._listwidget.currentItem()
+        name = str(current_item.text()) if current_item else ''
+        array = self.data[name] if name else None
```

Comment:
I tried with something like 
[1] arr4 = arr4.reshape(...)
which updated the title correctly. 
I guess I misunderstood when view_expr() is called.